atc/db/pipeline_pauser.go

```diff
@@ -0,0 +1,92 @@
+package db
+
+import (
+	"strconv"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/concourse/concourse/atc/db/lock"
+)
+
+//counterfeiter:generate . PipelinePauser
+type PipelinePauser interface {
+	PausePipelines(daysSinceLastBuild int) error
+}
+
+func NewPipelinePauser(conn Conn, lockFactory lock.LockFactory) PipelinePauser {
+	return &pipelinePauser{
+		conn:        conn,
+		lockFactory: lockFactory,
+	}
+}
+
+type pipelinePauser struct {
+	conn        Conn
+	lockFactory lock.LockFactory
+}
+
+func (p *pipelinePauser) PausePipelines(daysSinceLastBuild int) error {
+	activePipelines, err := getActivePipelines(p.conn, daysSinceLastBuild)
+	if err != nil {
+		return err
+	}
+
+	rows, err := pipelinesQuery.Where(sq.And{
+		sq.Eq{
+			"p.paused": false,
+		},
+		sq.NotEq{
```

Comment:
@taylorsilva Hmm does something like `Where(sq.Expr("SELECT p.id FROM pipelines AS p LEFT JOIN jobs AS j ON j.pipeline_id = p.id WHERE j.last_scheduled > CURRENT_DATE - ?::INTERVAL", strconv.Itoa(days) + " day)).` not work? This would help us not need to pass the pipelines over from one query to another, which can sometimes be slow if there are a lot active pipelines.